### ISSUE TITLE
Include multiple sets of versions not working

### ISSUE THREAD

**Issue by gander** [oven-sh/bun#4915]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 5.15.0-83-generic x86_64 x86_64

### What steps can reproduce the bug?

Prepare `package.json` dependencies:

```json
  "devDependencies": {
    "vitepress": "^1.0.0-rc.10 <1.0.0-rc.12 || >1.0.0-rc.12"
  }
```

OR

```shell
bun install "vitepress@^1.0.0-rc.10 <1.0.0-rc.12 || >1.0.0-rc.12"
```

### What is the expected behavior?

```
bun update v1.0.0 (822a00c4)
 + vitepress@1.0.0-rc.11
```

OR

```
bun add v1.0.0 (822a00c4)

 installed vitepress@1.0.0-rc.11 with binaries:
  - vitepress
```

### What do you see instead?

```
bun update v1.0.0 (822a00c4)
 + vitepress@1.0.0-rc.12
```

OR

```
bun add v1.0.0 (822a00c4)

 installed vitepress@1.0.0-rc.12 with binaries:
  - vitepress
```

### Additional information

Testing on [semver.npmjs.com](https://semver.npmjs.com/) this syntax should install one of:
+ `1.0.0-rc.10`
+ `1.0.0-rc.11`

![image](https://github.com/oven-sh/bun/assets/421597/23638e2b-ac90-4e26-9ea7-249cf76cf9b7)

**Comment by Jarred-Sumner** [comment]
@dylan-conway i think this was already fixed right?

**Comment by dylan-conway** [comment]
> @dylan-conway i think this was already fixed right?

Yes this was fixed around `~1.0.5`.

Closing this as fixed, please reopen or create a new issue if you continue to have problems with semver versions.
